src/coreclr/vm/jitinterface.cpp

```diff
@@ -10736,7 +10736,19 @@ void* CEEJitInfo::getHelperFtn(CorInfoHelpFunc    ftnNum,         /* IN  */
             dynamicFtnNum == DYNAMIC_CORINFO_HELP_DBL2ULNG_OVF)
         {
             Precode* pPrecode = Precode::GetPrecodeFromEntryPoint((PCODE)hlpDynamicFuncTable[dynamicFtnNum].pfnHelper);
-            _ASSERTE(pPrecode->GetType() == PRECODE_FIXUP);
+            PrecodeType precodeType = pPrecode->GetType();
+            if (precodeType == PRECODE_STUB)
```

Comment:
The precode type is effectively unpredictable. It depends on how MethodDescs are grouped into chunks and the type of the first MethodDesc in this chunk.

It means that whether or not this eager preparation is done would keep changing as methods are added/removed in CoreLib. I do not think we want unpredictable behavior like that.

I think we want to fix this by always creating fixup precode for temporary entrypoints: https://github.com/dotnet/runtime/pull/102372

Reply:
Also, when the tiered compilation is disabled, we generally pay for jump stubs or indirections for things that are not JITed yet. I do not think we should be making an exception for managed JIT helpers and prepare them eagerly.

Reply:
I see, thanks for the explanation!